File: dotnet/stack/Hosts/Service/Worker.cs
using Agience.SDK;

namespace Agience.Hosts.Service
{
    public class Worker : BackgroundService
    {
        private readonly ILogger<Worker> _logger;
        private SDK.Host _host;        

        public Worker(SDK.Host host, ILogger<Worker> logger)
        {
            _logger = logger;
            _host = host;           
            _host.AgentConnected += _host_AgentConnected;
        }

        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            _logger.LogInformation("Starting Host");

            await _host.RunAsync();

            _logger.LogInformation("Host Stopped");
        }


        private Task _host_AgentConnected(Agent agent)
        {
            _logger.LogInformation($"{agent.Name} Ready");

            return Task.CompletedTask;
        }
    }
}

File: core/hosts/python/services/worker.py
import logging
from core.host import Host
from core.agent import Agent


class Worker:
    def __init__(self, host: Host, logger: logging.Logger):
        self._logger = logger
        self._host = host
        self._host.agent_connected = self._host_agent_connected

    async def execute_async(self):
        self._logger.info("Starting Host")

        await self._host.run()

        self._logger.info("Host Stopped")

    async def _host_agent_connected(self, agent: Agent):
        self._logger.info(f"Agent {agent.name} Ready")
